File: 백준/Silver/13251. 조약돌 꺼내기/조약돌 꺼내기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int m = Integer.parseInt(br.readLine());

        int[] count = new int[m];
        int total = 0;

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            count[i] = Integer.parseInt(st.nextToken());
            total += count[i];
        }

        int k = Integer.parseInt(br.readLine());

        double answer = 0.0;
        for (int i = 0; i < m; i++) {
            if (count[i] >= k) {
                double probability = 1.0;

                for (int j = 0; j < k; j++) {
                    probability *= (double) (count[i] - j) / (total - j);
                }

                answer += probability;
            }
        }

        System.out.println(answer);
    }
}

File: 백준/Silver/13251. 조약돌 꺼내기/조약돌 꺼내기.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val m = br.readLine().toInt()
    val count = br.readLine().split(" ").map { it.toInt() }
    val total = count.sum()
    val k = br.readLine().toInt()

    var answer = 0.0
    for (i in 0 until m) {
        if (count[i] >= k) {
            var probability = 1.0

            for (j in 0 until k) {
                probability *= (count[i] - j).toDouble() / (total - j)
            }

            answer += probability
        }
    }

    println(answer)
}
